spec/builtin-trap-bash.test.sh

```diff
@@ -655,3 +655,175 @@ after=1
 
 ## N-I dash/mksh/ash STDOUT:
 ## END
+
+#### trap ERR with errtrace (-E)
+# ERR trap is not inherited by shell functions, sub processes, sub commands, ... unless the -o errtrace option is set
+set -E
+trap "echo ERR" ERR
+echo 0
+false
+echo 1
+cat <( date X ; echo 2 )
+echo $( date X ; echo 3 )
+{ date X ; echo 4 ; }
+{ date X ; echo 5 ; } & wait
+( date X ; echo 6 )
+f() { date X ; echo 7 ; } ; f
+x=$( date X ; echo 8 )
+true
+## status: 0
+## STDOUT:
+0
+ERR
+1
+ERR
+2
+ERR 3
+ERR
+4
+ERR
+5
+ERR
+6
+ERR
+7
+## END
+
+#### trap ERR without errtrace (+E)
+# ERR trap is not inherited by shell functions, sub processes, sub commands, ... unless the -o errtrace option is set
+set +E
+trap "echo ERR" ERR
+echo 0
+false
+echo 1
+cat <( date X ; echo 2 )
+echo $( date X ; echo 3 )
+{ date X ; echo 4 ; }
+{ date X ; echo 5 ; } & wait
+( date X ; echo 6 )
+f() { date X ; echo 7 ; } ; f
+x=$( date X ; echo 8 )
+true
+## status: 0
+## STDOUT:
+0
+ERR
+1
+2
+3
+ERR
+4
+5
+6
+7
+## END
+
+#### trap ERR LINENO
+# ERR trap is not inherited by shell functions, sub processes, sub commands, ... unless the -o errtrace option is set
+set +E
+trap 'echo ERR=$LINENO' ERR
+echo 0
+false
+true
+## status: 0
+## STDOUT:
+0
+ERR=4
+## END
+
+#### trap ERR NoLastFork
+trap 'echo ERR' ERR ; date X
+## status: 1
+## STDOUT:
+ERR
+## END
+
+#### trap ERR shadowing with errtrace (-E)
+# ERR trap is not inherited by shell functions, sub processes, sub commands, ... unless the -o errtrace option is set
+set -E
+trap "echo ERR" ERR
+false ; trap "echo ERRx" ERR ; false ; trap "echo ERR" ERR
+cat <( trap "echo ERR2" ERR ; date X )
```

Comment:
I think it's easier to follow if we break some of these cases up

I broke them up in to TWENTY cases in `spec/builtin-trap-err`

Reply:
Yes we can, it's just it takes much more time... and all 4 tests are near exactly the same, with and without errrtace, the same with shadowing...
This is just cosmetic we can do later once everything is working fine...